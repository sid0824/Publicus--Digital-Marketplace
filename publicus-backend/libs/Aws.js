import AWS from 'aws-sdk';
import {AccessKeyId, SecretAccessKey} from "../config/ConfigValues";

export const s3 = new AWS.S3({
  region: 'ap-southeast-2',
  accessKeyId: AccessKeyId,
  secretAccessKey: SecretAccessKey
});

export const ses = new AWS.SES({
  region: 'us-west-2'
});