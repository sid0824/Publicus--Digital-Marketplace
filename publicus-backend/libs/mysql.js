import mysql from 'mysql';
import {MySqlHost, MySqlPassword, MySqlUser} from "../config/ConfigValues";

//mysql connection to database
const connection = mysql.createConnection({
    host: MySqlHost,
    user: MySqlUser,
    password: MySqlPassword,
    database: "survey",
    multipleStatements: true
});

export const runQuery = async (query) => {
    return new Promise((resolve, reject) => {
        connection.query(query, (error, results) => {
            if(error) {
                reject(error);
            } else {
                resolve(results);
            }
        });
    });
};