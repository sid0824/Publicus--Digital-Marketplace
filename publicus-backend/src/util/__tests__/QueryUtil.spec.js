import { s3 } from "../../../libs/Aws";
import {
  checkAdmin,
  checkOwner,
  checkOwnerOrAdmin,
  checkOwnsForm,
  getSignedUrl,
  getUserNotificationSetting
} from "../QueryUtil";
import {runQuery} from "../../../libs/mysql";

jest.mock('../../../libs/Aws');
jest.mock("../../../libs/mysql");

describe('QueryUtil', () => {

  test(`should return receive notification setting`, async () => {
    runQuery.mockResolvedValueOnce([{ receiveNotifications: 1 }]);
    expect(runQuery.mock.calls[0]).toMatchSnapshot();
    expect(await getUserNotificationSetting('SURVEY_ID')).toEqual(1);
  });

  test(`should return if owner`, async () => {
    runQuery.mockResolvedValueOnce([{ userId: 'USER_ID' }]);
    expect(runQuery.mock.calls[0]).toMatchSnapshot();
    expect(await checkOwner('SURVEY_ID', 'USER_ID')).toBeTruthy();
  });

  test(`should return if admin`, async () => {
    runQuery.mockResolvedValueOnce([{ admin: 1 }]);
    expect(runQuery.mock.calls[0]).toMatchSnapshot();
    expect(await checkAdmin('USER_ID')).toBeTruthy();
  });

  test(`should return if owner or admin`, async () => {
    runQuery
      .mockResolvedValueOnce([{ userId: null }])
      .mockResolvedValueOnce([{ admin: 1 }]);
    expect(runQuery.mock.calls[0]).toMatchSnapshot();
    expect(runQuery.mock.calls[1]).toMatchSnapshot();
    expect(await checkOwnerOrAdmin('SURVEY_ID', 'USER_ID')).toBeTruthy();
  });

  test(`should get image s3 pre-signed url`, async () => {
    s3.getSignedUrl.mockImplementation((action, params, callback) => callback(null, 'http://signedurl.com'));
    const data = await getSignedUrl('FILE_PATH', 'FILE_TYPE');
    expect(data).toEqual('http://signedurl.com');
    expect(s3.getSignedUrl.mock.calls[0]).toMatchSnapshot();
  });

  test(`should return if a user owns a form`, async () => {
    runQuery.mockResolvedValueOnce([{ MOCK: 'VALUE' }]);
    const owns = await checkOwnsForm('USER_ID', 'SURVEY_ID');
    expect(owns).toBeTruthy();
    expect(runQuery.mock.calls[0]).toMatchSnapshot();
  });

});
