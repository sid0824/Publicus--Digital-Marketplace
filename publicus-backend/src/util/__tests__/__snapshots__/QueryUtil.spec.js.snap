// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryUtil should get image s3 pre-signed url 1`] = `
Array [
  "putObject",
  Object {
    "ACL": "public-read",
    "Bucket": "publicus-file-uploads",
    "ContentType": "FILE_TYPE",
    "Expires": 500,
    "Key": "FILE_PATH",
  },
  [Function],
]
`;

exports[`QueryUtil should return if a user owns a form 1`] = `
Array [
  "SELECT receiveNotifications
          FROM Surveys
          WHERE surveyId = 'SURVEY_ID'",
]
`;

exports[`QueryUtil should return if admin 1`] = `
Array [
  "SELECT receiveNotifications
          FROM Surveys
          WHERE surveyId = 'SURVEY_ID'",
]
`;

exports[`QueryUtil should return if owner 1`] = `
Array [
  "SELECT receiveNotifications
          FROM Surveys
          WHERE surveyId = 'SURVEY_ID'",
]
`;

exports[`QueryUtil should return if owner or admin 1`] = `
Array [
  "SELECT receiveNotifications
          FROM Surveys
          WHERE surveyId = 'SURVEY_ID'",
]
`;

exports[`QueryUtil should return if owner or admin 2`] = `
Array [
  "SELECT userId 
          FROM Surveys 
          WHERE surveyId = 'SURVEY_ID'",
]
`;

exports[`QueryUtil should return receive notification setting 1`] = `undefined`;
